decoder/label_candidates.go

```diff
@@ -81,6 +98,46 @@ func (d *Decoder) labelCandidatesFromDependentSchema(idx int, db map[schema.Sche
 	return candidates, nil
 }
 
+func generateRequiredFieldsSnippet(label string, bodySchema *schema.BodySchema, labelSchemas []*schema.LabelSchema, placeholder int, indentCount int) string {
+	snippetText := ""
+
+	if indentCount == 0 {
+		indentCount = 1
+	}
+	indent := strings.Repeat("\t", indentCount)
+
+	if len(labelSchemas) > 0 {
+		snippetText += fmt.Sprintf("%s\"", label)
+		for _, l := range labelSchemas[1:] {
+			snippetText += fmt.Sprintf(" \"${%d:%s}\"", placeholder, l.Name)
+			placeholder++
+		}
+		snippetText += " {\n"
+	}
+
+	attrNames := bodySchema.SortAttributes()
+	for _, attrName := range attrNames {
+		attr := bodySchema.Attributes[attrName]
+		if attr.IsRequired {
+			valueSnippet := snippetForExprContraint(uint(placeholder), attr.Expr)
+			snippetText += fmt.Sprintf("%s%s = %s\n", indent, attrName, valueSnippet)
+			placeholder++
+		}
+	}
+
+	blockNames := bodySchema.SortBlocks()
+	for _, blockName := range blockNames {
+		blockSchema := bodySchema.Blocks[blockName]
+		if blockSchema.MinItems > 0 {
+			snippetText += fmt.Sprintf("%s%s {\n", indent, blockName)
+			snippetText += generateRequiredFieldsSnippet("", blockSchema.Body, blockSchema.Labels, placeholder, indentCount+1)
```

Comment:
Correct me if I'm wrong here but I have a sense that this might not work for blocks of map type due to how we assume that all inner blocks never have labels? Do you mind adding a test with a block of `Type: schema.BlockTypeMap`? 🤔 

Blocks of map type generally always have a single label and while it's not super common for these blocks to have `MinItems` it's not impossible.

Reply:
Once I figure out how to model it, I will 😁